src/core/thread/mle_router.cpp

```diff
@@ -3729,11 +3731,14 @@ ThreadError MleRouter::AppendPendingDataset(Message &aMessage)
     ThreadError error = kThreadError_None;
     Tlv tlv;
 
+    VerifyOrExit(mNetif.GetPendingDataset().GetNetwork().GetSize() > 0,);
+
     tlv.SetType(Tlv::kPendingDataset);
-    tlv.SetLength(static_cast<uint8_t>(mNetif.GetPendingDataset().GetNetwork().GetSize()));
+    tlv.SetLength(static_cast<uint8_t>(mNetif.GetPendingDataset().GetNetwork().GetSize()) - sizeof(PendingTimestampTlv));
```

Comment:
It would be cleaner to just have the `Dataset` method append the entire TLV, including the TLV header.